pack/pack.go

```diff
@@ -46,24 +48,25 @@ type Event struct {
 	Links []httputil.Link `json:"links,omitempty"`
 }
 
-func (p *Pack) generateId() {
+type ok interface {
+	OK() error
+}
 
+func (p *Pack) generateId() {
 	id := p.Name
 	for _, k := range collections.SortedKeys(p.Labels) {
 		id += fmt.Sprintf(".%s.%s", k, p.Labels[k])
 	}
 	p.Id = id
 }
 
-func (p *Pack) setStatus() {
-	d := time.Since(p.LastSeen)
-	if d < 10 * time.Minute {
-		p.Status = "live"
-	} else if d < 24 * time.Hour {
-		p.Status = "warning"
-	} else {
-		p.Status = "critical"
+func (p *Pack) OK() error {
```

Comment:
This struct is used in couple of scenarios. One of them is to register pack another is to get pack. If the pack we are getting would have these links it would be a valid (OK) pack. Apart from that OK doesn't really mean a lot. I think what you are trying to do here is to validate that users don't pass certain links when posting packs. Add separate function to validate pack not as a method tho as pack with all this links is a valid pack the only thing we are trying to prevent is passing this links by user

Reply:
Well, I took this idea from this book: https://www.packtpub.com/gb/application-development/go-programming-blueprints-second-edition, but I'm happy to move it to a validate function.
At some point we will need to validate all the input data somehow for all these methods. I thought this could be a clean way to check all required fields.

Reply:
I came across this article https://medium.com/@apzuk3/input-validation-in-golang-bc24cdec1835

The main thing from my point of view is  validation is a separate process. Pack entity is used by the whole package it is not only used for a user input. If this entity would be only used for user input than it would be OKish to have maybe a method called `IsValid` `HasValidLinks` etc... still not ideal.

In our case it works well with just separate function to validate links. It is very clean and intentions are very obvious. Makes it easy to maintain e.g. replace with yaml decoder, toml decoder. Where validation according to some business logic is a separate process. When we will need a solution for validation then I guess we have to add something but it certainly should not be done with OK interface